modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/discovery/ShreddedType.scala

```diff
@@ -111,30 +115,41 @@ object ShreddedType {
                         name: String,
                         model: Int,
                         shredJob: Semver,
-                        shredProperty: LoaderMessage.ShreddedType.ShredProperty) {
+                        snowplowEntity: LoaderMessage.SnowplowEntity) {
     def toCriterion: SchemaCriterion = SchemaCriterion(vendor, name, "jsonschema", model)
   }
 
   /**
     * Transform common shredded type into loader-ready. TSV is isomorphic and cannot fail,
     * but JSONPath-based must have JSONPath file discovered - it's the only possible point of failure
     */
-  def fromCommon[F[_]: Monad: Cache: AWS](
+  def fromCommonShred[F[_]: Monad: Cache: AWS](
     base: S3.Folder,
     shredJob: Semver,
     region: String,
     jsonpathAssets: Option[S3.Folder],
-    commonType: LoaderMessage.ShreddedType
+    commonType: TypesInfo.Shredded.Type
```

Comment:
Using covariant type `commonType: TypesInfo` would allow to keep` fromCommon` without branching the code downstream.

Reply:
But, it is taking `TypesInfo.Shredded.Type` as argument, not `TypesInfo`. And, `TypesInfo.Shredded.Type` and `TypesInfo.WideRow.Type` doesn't have common super type. May I ask if I am missing something ?

Reply:
I guess I understand what you mean, I will make necessary change.